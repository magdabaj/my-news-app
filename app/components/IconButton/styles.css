.iconButton { /* stylelint-disable */
    text-align: center;
    cursor: pointer;
    margin-right: 15px;
}
