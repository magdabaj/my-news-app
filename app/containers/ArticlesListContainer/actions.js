/*
 *
 * ArticlesListContainer actions
 *
 */

import {
  GET_ALL_ARTICLES,
  GET_ARTICLES_FOR_TAG,
  GET_ARTICLES_FOR_TAG_FAILED,
  GET_ARTICLES_FOR_TAG_SUCCESS, START_ADD
} from "./constants";

export const getAllArticles = () => ({
  type: GET_ALL_ARTICLES
});

export const getArticlesForTag = tagId => ({
  type: GET_ARTICLES_FOR_TAG,
  tagId,
});

export const getArticlesForTagSuccess = articles => ({
  type: GET_ARTICLES_FOR_TAG_SUCCESS,
  articles,
});

export const getArticlesForTagFailed = error => ({
  type: GET_ARTICLES_FOR_TAG_FAILED,
  error,
});

export const startAdd = tagName => ({
  type: START_ADD,
  tagName,
});