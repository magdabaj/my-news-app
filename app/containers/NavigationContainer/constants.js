/*
 *
 * NavigationContainer constants
 *
 */

export const GET_TAGS = "app/NavigationContainer/GET_TAGS";
export const GET_TAGS_SUCCESS = "app/NavigationContainer/GET_TAGS_SUCCESS";
export const GET_TAGS_FAILED = "app/NavigationContainer/GET_TAGS_FAILED";

export const GET_ARTICLES_BY_TAG = "app/NavigationContainer/GET_ARTICLES_BY_TAG";
