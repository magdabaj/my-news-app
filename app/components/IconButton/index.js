/**
 *
 * IconButton
 *
 */

import React from "react";
import PropTypes from 'prop-types';
// import styled from 'styled-components';
import classNames from 'classnames';
import FontAwesome from 'react-fontawesome';
import './styles.css';

const IconButton = ({icon, onClick, buttonClass, iconClass}) => {
  return <div
      className={'iconButton'}
      onClick={onClick}
  >
    <FontAwesome
        className={iconClass}
        name={icon}
    />
  </div>;
};

IconButton.propTypes = {
  icon: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  buttonClass: PropTypes.string,
  iconClass: PropTypes.string,
};

export default IconButton;
