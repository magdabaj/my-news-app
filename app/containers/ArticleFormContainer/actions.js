/*
 *
 * ArticleFormContainer actions
 *
 */

import {ADD_ARTICLE, ADD_ARTICLE_CANCELED, ADD_ARTICLE_ERROR, ADD_ARTICLE_SUCCESS} from "./constants";

export const addArticle = (article, tagId, user) => ({
  type: ADD_ARTICLE,
  article,
  tagId,
  user
});

export const addArticleSuccess = article => ({
  type: ADD_ARTICLE_SUCCESS,
  article,
});

export const addArticleError = error => ({
  type: ADD_ARTICLE_ERROR,
  error,
});

export const addArticleCanceled = () => ({
  type: ADD_ARTICLE_CANCELED,
});