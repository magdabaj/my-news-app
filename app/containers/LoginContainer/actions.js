/*
 *
 * LoginContainer actions
 *
 */

import {CANCEL_LOGIN, LOGIN, LOGIN_FAILED, LOGIN_SUCCESS} from "./constants";

export const login = (email, password) => ({
  type: LOGIN,
  email,
  password,
});

export const loginSuccess = user => ({
  type: LOGIN_SUCCESS,
  user,
});

export const loginFailed = error => ({
  type: LOGIN_FAILED,
  error,
});

export const cancelLogin = () => ({
  type: CANCEL_LOGIN,
});