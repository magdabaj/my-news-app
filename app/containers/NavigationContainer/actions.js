/*
 *
 * NavigationContainer actions
 *
 */

import { GET_TAGS, GET_ARTICLES_BY_TAG, GET_TAGS_FAILED, GET_TAGS_SUCCESS } from "./constants";

export const getTags = () => ({
  type: GET_TAGS
});

export const getTagsSuccess = tags => ({
  type: GET_TAGS_SUCCESS,
  tags
});

export const getTagsFailed = error => ({
  type: GET_TAGS_FAILED,
  error
});

export const getArticlesByTag = tagId => ({
  type: GET_ARTICLES_BY_TAG,
  tagId,
});
