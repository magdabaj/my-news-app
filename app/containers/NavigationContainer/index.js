/**
 *
 * NavigationContainer
 *
 */

import React, { memo, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { compose } from "redux";

import { useInjectSaga } from "utils/injectSaga";
import { useInjectReducer } from "utils/injectReducer";
import makeSelectNavigationContainer, {
  makeSelectIsDrawerOpen,
  makeSelectSelectedTag,
  makeSelectTags
} from "./selectors";
import reducer from "./reducer";
import saga from "./saga";
import { getTags, selectTag, toggleDrawer} from './actions';
import Navigation from "../../components/Navigation";

export function NavigationContainer({...props }) {
  useInjectReducer({ key: "navigationContainer", reducer });
  useInjectSaga({ key: "navigationContainer", saga });

  useEffect( () => {
    if (props.tags.length === 0) {
      props.getTags();
    }
  }, []);

  return <Navigation {...props} />;
}

// NavigationContainer.propTypes = {
//   tags: PropTypes.array.isRequired,
//   getTags: PropTypes.func.isRequired,
//   getArticlesByTag: PropTypes.func.isRequired,
//   selectTag: PropTypes.func.isRequired,
//   toggleDrawer: PropTypes.func.isRequired,
//   selectedTag: PropTypes.object,
//   isDrawerOpen: PropTypes.bool.isRequired,
// };

const mapStateToProps = createStructuredSelector({
  navigationContainer: makeSelectNavigationContainer(),
  tags: makeSelectTags(),
  selectedTag: makeSelectSelectedTag(),
  isDrawerOpen: makeSelectIsDrawerOpen(),
});

function mapDispatchToProps(dispatch) {
  return {
    getTags: () => dispatch(getTags()),
    selectTag: selectedTag => dispatch(selectTag(selectedTag)),
    toggleDrawer: () => dispatch(toggleDrawer()),
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps
);

export default compose(
  withConnect,
  memo
)(NavigationContainer);
