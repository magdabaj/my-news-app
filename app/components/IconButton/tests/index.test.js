/**
 *
 * Tests for IconButton
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from "react";
import { render } from "react-testing-library";
import 'jest-dom/extend-expect'; // add some helpful assertions

import IconButton from "../index";
import {toggleDrawer} from "../../../containers/NavigationContainer/actions";

jest.mock("../../../containers/NavigationContainer/actions");

describe("<IconButton />", () => {

  const icon = "bars";
  const buttonCLass = "iconButton";
  const iconClass = 'icon';

  it("Expect to not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error");
    render(<IconButton onClick={toggleDrawer} icon={icon} buttonClass={buttonCLass} iconClass={iconClass}/>);
    expect(spy).not.toHaveBeenCalled();
  });

  it("Expect to have additional unit tests specified", () => {
    expect(true).toEqual(true);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild }
    } = render(<IconButton onClick={toggleDrawer} icon={icon} buttonClass={buttonCLass} iconClass={iconClass}/>);
    expect(firstChild).toMatchSnapshot();
  });
});
