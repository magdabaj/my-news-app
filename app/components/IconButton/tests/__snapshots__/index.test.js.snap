// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> Should render and match the snapshot 1`] = `
<div
  class="iconButton"
>
  <span
    aria-hidden="true"
    class="fa fa-bars icon"
  />
</div>
`;
